plugins/poll/app/models/poll.rb

```diff
@@ -40,7 +41,10 @@ def is_closed?
   end
 
   def can_see_results?(user)
-    always? || is_closed? || (on_vote? && has_voted?(user))
+    return true if always?
+    return user&.staff? if staff_only?
```

Comment:
Since we want to return a `boolean`, we might want to change this line

```suggestion
    return !!user&.staff? if staff_only?
```